README.md

```diff
@@ -2,8 +2,147 @@
 
 [![Build Status](https://travis-ci.org/hmrc/file-transmission.svg)](https://travis-ci.org/hmrc/file-transmission) [ ![Download](https://api.bintray.com/packages/hmrc/releases/file-transmission/images/download.svg) ](https://bintray.com/hmrc/releases/file-transmission/_latestVersion)
 
-This is a placeholder README.md for a new repository
+Microservice that facilitates transmission of files between MDTP and MDG.
+
+Whenever service on MDTP platform wants to send bundle (batch) of files to MDG it will need to use this service. The file-transmission
+service notifies MDG that files prepared by service on MDTP are ready to be processed. Please note that file-transmission service 
+do not transfer file contents. The service just passes data about file location. The file content has to be provided by other service (eg. upscan)
+
+# Typical usage scenario
+
+- Consuming service requests upload of the file from user using `upscan` service
+- `Upscan` notifies consuming service that the file has been uploaded and is ready to be processed - giving consuming 
+service the URL that allows to download the file.
+- Consuming service ensures that all required files have been uploaded by the user
+- Consuming service uses file-transmission service to notify MDG that files are ready to be processes
+- MDG asynchronously uses the file
+
+## Service usage
+
+### Whitelisting
+
+In order to initiate an upload the consuming service must be whitelisted by file-transmission. See the 'Whitelisting client services' section further down in this document.
+
+### Requesting a file transmission
+
+The basic unit of work for file-transmission service is a batch that might consist of one or more files.
+
+Information about every file in the batch is passed to the file-transmission service as a separate POST request. 
+There is no need to make additional call to create a batch or to notify that data about all files in the batch have 
+been provided.
+
+Assuming the consuming service is whitelisted, it makes a POST request to the `/file-transmission/request` endpoint. 
+The request contains data about batch, details of the file and any additional metadata which consuming service wants
+to pass to MDG. The request should also contain callback URL that will be used to asynchronously notify consuming service 
+that MDG has processed the request.
+
+The following information has to pe passed within request:
+- Batch identifier
+- Number of files in the batch
+- `callbackUrl` that will be used by MDG to notify upstream service about result of batch processing.
+- `journeyName` / `journeyVersion` - information for MDG about meaning of files in the batch, determining what process should be invoked.
+- `requestTimeoutInSeconds` - time describing how long file-transmission service will try to deliver file details to MDG. 
```

Comment:
Maybe worth describing here what happens and what the service should expect after this time duration has expired (e.g a call from file-transmission, or the absence of a call by this point). Again, maybe to be clarified later.